From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Helvetica Volubi <suisuroru@blue-millennium.fun>
Date: Fri, 3 Oct 2025 19:43:36 +0800
Subject: [PATCH] Modify merge ItemEntity logic

Add followTickSequenceMerge to modify merge items, due to Paper's modification of the merge radius, when the merge radius is large and stacks containing many items get stuck in an unexpected position, individual items may never reach their destination. This configuration option is added to fix this behavior.

diff --git a/net/minecraft/world/entity/item/ItemEntity.java b/net/minecraft/world/entity/item/ItemEntity.java
index 121e18a1825666e19a4e885735caf7b13040e64a..b4384bc4518a58f275a8635a8b918d1f2b95ac11 100644
--- a/net/minecraft/world/entity/item/ItemEntity.java
+++ b/net/minecraft/world/entity/item/ItemEntity.java
@@ -288,7 +288,7 @@ public class ItemEntity extends Entity implements TraceableEntity {
         ItemStack thisItemStack = this.getItem();
         ItemStack otherItemStack = other.getItem();
         if (Objects.equals(this.target, other.target) && areMergable(thisItemStack, otherItemStack)) {
-            if (otherItemStack.getCount() < thisItemStack.getCount()) {
+            if (org.leavesmc.leaves.LeavesConfig.modify.followTickSequenceMerge || otherItemStack.getCount() < thisItemStack.getCount()) { // Leaves - add follow Tick Sequence Merge, see Paper#13073
                 merge(this, thisItemStack, other, otherItemStack);
             } else {
                 merge(other, otherItemStack, this, thisItemStack);
